### ISSUE TITLE
Fila de publicação: filtro de campo aberto

### ISSUE THREAD

**Issue by sahrosa** [servicosgovbr/editor-de-servicos#387]
Cartão criado a partir do #339

- - - 

- [ ] Filtro de campo aberto deverá buscar por todos os conteúdos (áreas de interesse, órgãos, serviços, páginas especiais) 

<img width="1088" alt="3" src="https://cloud.githubusercontent.com/assets/2932644/10221974/e3e18214-6827-11e5-83a0-a97af66768c5.png">

**Comment by kchnr** [comment]
Vai buscar no conteúdo de cada página, ou em todos os tipos de página?

**Comment by sahrosa** [comment]
Acredito que apenas no título da página, @kchnr, pois é um filtro para acesso mais rápido a uma página (seja serviço, seja órgão, área de interesse ou página especial etc).

**Comment by glauberramos** [comment]
@sahrosa isso já não está funcionando?

**Comment by sahrosa** [comment]
A princípio sim @glauberramos! 

@almeidafab @mformiga @gustavo-freitas, este cartão era apenas para confirmar se o filtro estava funcionando, e pelo que testamos está ok. Podemos passar para homologação?

**Comment by cevmelo** [comment]
Bom dia pessoal,

ao pesquisar pela palavra "Acessibilidade" com todos os filtros desligados é retornada lista contendo por ex: o serviço "Atividade Rural". A palavra pesquisada não está contida no título do serviço ou nos outros campos do mesmo. 

Ao pesquisar a palavra "Secretaria" somente com o filtro "Página de Órgão" ligado é retornada lista contendo por ex: o órgão "Defensoria Pública da União (DPU)", a palavra pesquisada também não está contida no conteúdo da página.

Favor verificar.

**Comment by almeidafab** [comment]
@sahrosa o @cevmelo está me ajudando com os testes agora

**Comment by sahrosa** [comment]
Legal @almeidafab @cevmelo!

Vamos verificar!

**Comment by gustavo-freitas** [comment]
Verificamos e entendemos que essa busca deve ser refinada @cevmelo.

Entretanto, o seguinte ponto serão considerado na implementação:

- Os primeiros resultados exibidos devem ter relação direta com o termo pesquisado.

**Comment by sahrosa** [comment]
Pessoal,
fizemos mudanças que impactam significativamente o filtro do Editor. A mudança principal é na quantidade de caracteres que podem estar errados (fora de ordem, por exemplo) que antes era de 80% de erro e passou para 50% de erro. 

@almeidafab @gustavo-freitas vocês podem verificar se está ok para passarmos para a homologação? :)

**Comment by gustavo-freitas** [comment]
@sahrosa @kchnr o cenário que o @cevmelo descreveu não acontece mais. Além disso, os resultados da busca estão bem mais coerentes.

Entretanto, notei o seguinte cenário (acredito que não tem relação direta com esse cartão):

1. Quando filtro de orgão é utilizado (página de órgãos)
2. Busca não consegue identificar corretamente alguns itens na lista de órgãos (por exemplo: caso usuário busque por 'Justiça', nada será retornado, mas duas ocorrências deveriam ser encontradas)

Esse cenário que descrevi é em Alpha. Considerando que os cartões dos filtros já foram homologados e não validamos a busca neles, acredito que um novo cartão possa nos ajudar a resolver esse possível problema (bug). O que vocês acham?

**Comment by oliviagj** [comment]
Sendo o caso que você mencionou o "Justiça", podemos testar a hipótese de
ser um problema com o caractere especial (ç), o que você acha,
@gustavo-freitas?

**Comment by gustavo-freitas** [comment]
@oliviagj claro! Acabei de reproduzir novamente aqui. Na realidade, não acontece somente com caractere especial, mas qualquer palavra que não seja a primeira ou segunda de um nome. Por exemplo: Ministério da Saúde é exibido na lista, usuário procura por 'Saúde', nenhuma ocorrência é encontrada. Isso talvez seja um problema maior. @kchnr @sahrosa sugestões?

**Comment by sahrosa** [comment]
pois é @gustavo-freitas. Testando aqui por "ministério justiça" foi encontrado normalmente o órgão, mas buscando por "justiça" e "justiça ministério".

Pelo que repeti com outros itens como serviço "Serviço de teste" o problema é com a segunda palavra e não está relacionado com caractere especial. @kchnr, isso pode ter a ver com os parâmetros que vimos ontem?

**Comment by gustavo-freitas** [comment]
@sahrosa combinei com o @kchnr que vou investigar isso! Tá comigo.

**Comment by gustavo-freitas** [comment]
Busca refinada de acordo com feedback de @oliviagj @sahrosa e @kchnr. Cenários que @cevmelo descreveu não acontecem mais.

- Quando usuário pesquisa  pela palavra "Acessibilidade", como esperado, somente um resultado é retornado
- Quando usuário pesquisa pela palavra "Secretaria", com o filtro "Página de Órgão" ligado, é retornada uma lista contendo somente ocorrências que possuem "Secretaria"
- Quando usuário pesquisa por 'Justiça', 'Ministério da Justiça é retornado
- Busca funciona corretamente com ou sem o uso de filtros

**Comment by sahrosa** [comment]
Muito bom @gustavo-freitas! :+1:

**Comment by gustavo-freitas** [comment]
Revisado com @almeidafab e @mformiga. Passando direto para produção.
